src/lib/xmlparse.rs

```diff
@@ -126,9 +126,151 @@ use alloc_wg::alloc::{AllocRef, BuildAllocRef, DeallocRef, NonZeroLayout, Reallo
 use alloc_wg::boxed::Box;
 
 use std::collections::{hash_map, HashMap};
+use std::convert::TryInto;
 use std::mem;
+use std::ops;
 use std::ptr::{self, NonNull};
 
+#[derive(Copy, Clone)]
+pub struct ExpatBufRef<'a, T = c_char>(&'a [T]);
```

Comment:
What are the benefits of using `ExpatBufRef` instead of a slice directly?

Reply:
Constructors and associated methods. I guess I could implement a trait on &'a [T], but this seemed cleaner.